### PR TITLE
Upload Windows Terminal 2.0 roadmap

### PR DESCRIPTION
**Pull request by cinnamon-msft**
## Summary of the Pull Request
Upload the roadmap for Windows Terminal 2.0 and link to it on the README.

 
## PR Checklist
* [ ] Closes #xxx
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [x] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

### CODE REVIEW

File: doc/terminal-v2-roadmap.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+| --- | --- | --- |
+| 2 weeks | Dev Work<br/> <ul><li>Fixes / Features for future Windows Releases</li><li>Fixes / Features for Windows Terminal</li></ul> |  Release to Internal Selfhosters at end of week 2 |
+| 1 week | Quality & Stability<br/> <ul><li>Bug Fixes</li><li>Perf & Stability</li><li>UI Polish</li><li>Tests</li><li>etc.</li></ul>| Push to Microsoft Store at end of week 3 |
+| 1 week | Release <br/> <ul><li>Available from [Microsoft Store](https://aka.ms/terminal) & [GitHub Releases](https://github.com/microsoft/terminal/releases)</li><li>Release Notes & Announcement Blog published</li><li>Engineering System Maintenance</li><li>Community Engagement</li><li>Docs</li><li>Future Milestone Planning</li></ul> | Release available from Microsoft Store & GitHub Releases |
+
+## Terminal Roadmap / Timeline
+
+| Milestone end date | Milestone Name | Preview Release Blog Post |
+| ------------------ | -------------- | ------------------------- |
+| 2020-06-30 | [1.1](https://github.com/microsoft/terminal/milestone/24) in Windows Terminal Preview |
```
Comment (zadjii-msft) [inline doc/terminal-v2-roadmap.md:21]: nit: maybe add a note that "the specific end dates are subject to change / are a rough estimate"

File: doc/terminal-v2-roadmap.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+## 2.0 Scenarios
+
+The following are a list of the key scenarios we're aiming to deliver for Terminal 2.0.
+
+> 👉 Note: There are many other features that don't fit within 2.0, but will be re-assessed and prioritized for 3.0, the plan for which will be published in 2021.
+
+| Priority\* | Scenario | Description/Notes |
+| ---------- | -------- | ----------------- |
+| 0 | Settings UI | A user interface that connects to settings.json. This provides a way for people to edit their settings without having to edit a JSON file.<br><br>Issue: [#1564] |
```
Comment (zadjii-msft) [inline doc/terminal-v2-roadmap.md:50]: I'm not sure github will auto-link issues in `[#1564]` format within a Markdown doc if you don't add the following to the bottom of the doc:
```md
[#1564]: https://github.com/microsoft/terminal/issues/1564
```

(for each and every one)

Comment (cinnamon-msft) [reply doc/terminal-v2-roadmap.md:50]: Oh sad 😥

File: doc/terminal-v2-roadmap.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
+# Terminal 2.0 Roadmap
+
+## Overview
+
+This document outlines the roadmap towards delivering Windows Terminal 2.0 by spring 2021.
```
Comment (carlos-zamora) [inline doc/terminal-v2-roadmap.md:5]:
```suggestion
This document outlines the roadmap towards delivering Windows Terminal 2.0 by Spring 2021.

```

File: doc/terminal-v2-roadmap.md (hunk 2 above)
Comment (carlos-zamora) [reply doc/terminal-v2-roadmap.md]: image
Try this, maybe?
Reference: https://guides.github.com/features/mastering-markdown/

File: doc/terminal-v2-roadmap.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+> 👉 Note: There are many other features that don't fit within 2.0, but will be re-assessed and prioritized for 3.0, the plan for which will be published in 2021.
+
+| Priority\* | Scenario | Description/Notes |
+| ---------- | -------- | ----------------- |
+| 0 | Settings UI | A user interface that connects to settings.json. This provides a way for people to edit their settings without having to edit a JSON file.<br><br>Issue: [#1564] |
+| 0 | Command palette | A popup menu to list possible actions and commands.<br><br>Issues: [#5400], [#2046]<br>Spec: [#2193] |
+| 1 | Tab tear-off | The ability to tear a tab out of the current window and spawn a new window or attach it to a separate window.<br><br>Issue: [#1256]<br>Spec: [#2080] |
+| 1 | Clickable links | Hyperlinking any links that appear in the text buffer. When clicking on the link, the link will open in your default browser.<br><br>Issue: [#574] |
+| 1 | Default terminal | If a command-line application is spawned, it should open in Windows Terminal (if installed)\*.<br>\* Or your preferred terminal<br><br>Issue: [#492]<br>Spec: [#2080] |
```
Comment (carlos-zamora) [inline doc/terminal-v2-roadmap.md]:
```suggestion
| 1 | Default terminal | If a command-line application is spawned, it should open in Windows Terminal (if installed) or your preferred terminal<br><br>Issue: [#492]<br>Spec: [#2080] |
```
A bit unnecessary.

File: doc/terminal-v2-roadmap.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+| 1 | Overall theme support | Tab coloring, title bar coloring, pane border coloring, pane border width, definition of what makes a theme<br><br>Issue: [#3327]<br>Spec: [#5772] |
+| 1 | Open tab as admin/other user | Open tab in existing Windows Terminal instance as admin (if Terminal was run unelevated) or as another user.<br><br>Issue: [#5000] |
+| 1 | Traditional opacity | Don't add the acrylic blur.<br><br>Issue: [#603] |
```
Comment (carlos-zamora) [inline doc/terminal-v2-roadmap.md]: Rewrite this description. Telling me what a feature doesn't do doesn't actually tell me what it does hahaha.

File: doc/terminal-v2-roadmap.md (hunk 2 above)
Comment (cinnamon-msft) [reply doc/terminal-v2-roadmap.md:53]: No bueno :(

**Comment by cinnamon-msft** [comment]
> A few things:
> 
> * You included the implementation in one of the issues. Should this be expected for all of them as we go? Feels weird to include one and not the others. Especially if somebody comes across this doc a few months from now and sees that it's out of date...
> * Can you link the 2.0 milestone somehow? Or at least mention it?

I plan on updating this with implementation links as well as the release blog posts.
The 2.0 milestone is linked throughout the doc.

**Comment by carlos-zamora** [comment]
@msftbot make sure @DHowett signs off on this

**Pull request merged**